src/ListGroup.js

```diff
@@ -8,24 +8,37 @@ const propTypes = {
   flush: PropTypes.bool,
   className: PropTypes.string,
   cssModule: PropTypes.object,
+  horizontal: PropTypes.oneOfType([PropTypes.bool, PropTypes.string])
 };
 
 const defaultProps = {
   tag: 'ul'
 };
 
+const getHorizontalClass = horizontal => {
+  if (horizontal === false) {
+    return false;
+  } else if (horizontal === true || horizontal === "xs") {
+    return "list-group-horizontal";
+  }
+  return `list-group-horizontal-${horizontal}`;
+};
+
 const ListGroup = (props) => {
   const {
     className,
     cssModule,
     tag: Tag,
     flush,
+    horizontal,
     ...attributes
   } = props;
   const classes = mapToCssModules(classNames(
     className,
     'list-group',
-    flush ? 'list-group-flush' : false
+    // list-group-horizontal cannot currently be mixed with list-group-flush
+    // we only try to apply horizontal classes if flush is false
+    flush ? 'list-group-flush' : getHorizontalClass(horizontal)
```

Comment:
When `flush` and `horizontal` are not provided, they are `undefined`. This code would go into `getHorizontalClass(undefined)`. Since `undefined === false` is `false`, it would go to 
```
return `list-group-horizontal-${horizontal}`;
```
and produce "list-group-horizontal-undefined".

Reply:
Oops, good catch thanks. I had meant to add a default value for `horizontal`. Added that now, are you happy with that as a fix?